pandas/core/series.py

```diff
@@ -2291,7 +2291,9 @@ def corr(self, other, method="pearson", min_periods=None) -> float:
             f"'{method}' was supplied"
         )
 
-    def cov(self, other, min_periods=None) -> float:
+    def cov(
+        self, other: "Series", min_periods: Optional[int] = None, ddof: int = 1
```

Comment:
```suggestion
        self, other: "Series", min_periods: Optional[int] = None, ddof: Optional[int] = 1
```

Reply:
Based on the official documentation https://docs.python.org/3/library/typing.html
it seems that the Optional should not be applied to optional parameters with a default (like ddof).

Reply:
What the docs mean is that an argument with a default shouldn't be given an Optional type _because providing the value is optional_. It should have an Optional type if None is a possible value:

> On the other hand, if an explicit value of None is allowed, the use of Optional is appropriate, whether the argument is optional or not.

Reply:
I see your point. Numpy is implementing the protection for None and it's actually what is present in the test case. Made Optional